src/users/users.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { User } from './model-user';
+import { CreateUsersDto } from './DTO/create-user.dto';
+import {v4 as uuid} from 'uuid';
+
+
+
+@Injectable()
+export class UsersService {
+  private users: User[]= [];
+
+  getAllUsers(): User[] {
+    return this.users
+  }
+
+  getUserById(id:string): User{
+    const user = this.users.find( user => user.id === id);
+
+    if (!user) {
+     throw new NotFoundException();
+      }
+   return user;
+  }
+
+  createUser(createUsersDto: CreateUsersDto): User{
+    const {fullName, email, password} = createUsersDto
+
+    // const salt= bcrypt.genSalt(10);
```

Comment:
Why the password is not being hashed? another think is that you are returning the password to the user, so, the same with the getAll and getById method, return a UserResponseDto.